quantum/process_keycode/process_auto_shift.h

```diff
@@ -22,13 +22,34 @@
 #    define AUTO_SHIFT_TIMEOUT 175
 #endif
 
+#define IS_LT(kc) ((kc) >= QK_LAYER_TAP && (kc) <= QK_LAYER_TAP_MAX)
+#define IS_MT(kc) ((kc) >= QK_MOD_TAP && (kc) <= QK_MOD_TAP_MAX)
+#define IS_RETRO(kc) (IS_MT(kc) || IS_LT(kc))
+#define DO_GET_AUTOSHIFT_TIMEOUT(keycode, record, ...) record
+// clang-format off
+#define get_autoshift_timeout(...) _Generic(        \
+        (DO_GET_AUTOSHIFT_TIMEOUT(__VA_ARGS__, 0)), \
+            keyrecord_t *: (get_autoshift_timeout), \
+            default: get_generic_autoshift_timeout  \
+)(__VA_ARGS__)
```

Comment:
Maintainers, thoughts? Ugly, and not very C-like, but this pull isn't a breaking change at the moment and I would like to keep it that way. I also don't want Auto Shift to have the single `PER_KEY` option that isn't called `get_[value]`. This makes `get_autoshift_timeout` either call the user function, or `get_generic_autoshift_timeout` if not passed the keycode and record pointer. The only impact on the user is that their `get_autoshift_timeout` definition must have parentheses around it (noted in the docs).

Reply:
Definitely ugly.

This should be a c function, rather than a define.

Reply:
Alright, I will change it. I suppose something changing this much would probably go through the breaking change process, anyway (I believe the keyrepeat one did), and it's not like it's hard to quick change a single function name when it breaks people's keymaps.